### DISCUSSION TITLE
トラックボールのみ動作しない

### DISCUSSION THREAD

**Discussion by sinpost2** [Yowkees/keyball#274] [Q&A] [answered]
keyball61を購入、7月上旬に組み上げたもののトラックボールのみ動作しません。
原因として考えられる事や解決策を教えていただけますでしょうか。

現状
・トラックボールは一度も反応したことがありません。
・OLEDにはトラックボールの移動量も表示されない（全て0のまま）。
・キーボードやマウスクリック、LEDの動作等は正常。
・左手側にball。

対策？
・おそらく、下記の対応が相応しいのではないか？と思いますが、
　当方が初心者のため確証を持てずにおります。

　トラックボールが動きません #51
　https://github.com/Yowkees/keyball/discussions/51

何卒よろしくお願いいたします。

<img width="1036" alt="スクリーンショット 2023-07-14 7 13 45" src="https://github.com/Yowkees/keyball/assets/57223126/a3e8a5f8-6a23-4af6-b2e9-493b214133ee">
<img width="800" alt="スクリーンショット 2023-07-14 7 14 36" src="https://github.com/Yowkees/keyball/assets/57223126/06554a63-8ebb-48f4-a679-01c9f47fb67d">
<img width="620" alt="スクリーンショット 2023-07-14 7 16 07" src="https://github.com/Yowkees/keyball/assets/57223126/36a28873-6e4f-4fbb-8164-04adffd8706e">
<img width="443" alt="スクリーンショット 2023-07-14 7 13 28" src="https://github.com/Yowkees/keyball/assets/57223126/ca051cfb-dbdc-4655-9fab-2ee3e296ac6c">
<img width="288" alt="スクリーンショット 2023-07-14 7 15 45" src="https://github.com/Yowkees/keyball/assets/57223126/71631c92-4964-4f41-a82b-380b1d66018d">

**Comment by Yowkees** [comment, marked as answer]
ご連絡ありがとうございます。
お写真見せていただきましたが、半田付け忘れ等ありませんし、全体的にキレイに半田付けできているので動きそうですがどうしてでしょう。
ProMicroの初期不良の可能性もありますので、まず、左右のProMicroを入れ替えたらどうなるかご確認いただけますでしょうか。
お手数ですがよろしくお願いいたします。

**Reply by sinpost2** [reply]
早速のご返信ありがとうございます。

ご指摘のとおり左右のProMicroを入れ替えたら、トラックボールが動作しました。
やっとKeyballの本領を味わうことができました。
ありがとうございます。

ただし、
・代わりに右側のキーボードが一つ反応しなくなりました。
<img width="1048" alt="スクリーンショット 2023-07-15 14 57 27" src="https://github.com/Yowkees/keyball/assets/57223126/097bf762-03de-485c-8ada-578843739535">

原因を調べるためよくみると、
右側に移動させたProMicroのピンヘッダの一本が短くなっていました。
![IMG_4228](https://github.com/Yowkees/keyball/assets/57223126/e8a8e589-2f34-44bd-95bc-6d548f64e24d)

加えて、問題のあった左側のソケットにProMicroが入り切らず少し数ミリ浮いていました。

ここから推測するに、もしかするとこの一本が欠けてソケットに残りっぱなしなのかもしれません。

新品に入れ替えるなりして対応しようと思います。

Keyball使えること嬉しいです。引き続きよろしくお願いします。

**Comment by Yowkees** [comment]
ご調査ありがとうございます。
ピンソケットのピン浮きが原因のようですね。Keyball同梱のピンソケットの方が高さも抑えられますので、そちらに交換いただければ幸いです。
お手数ですがご対応お願いいたします。
